api/agent/uniter/unit.go

```diff
@@ -177,25 +178,66 @@ func (s *Unit) Watch(ctx context.Context) (watcher.NotifyWatcher, error) {
 // WatchRelations returns a StringsWatcher that notifies of changes to
 // the lifecycles of relations involving u.
 func (u *Unit) WatchRelations(ctx context.Context) (watcher.StringsWatcher, error) {
-	var results params.StringsWatchResults
-	args := params.Entities{
-		Entities: []params.Entity{{Tag: u.tag.String()}},
-	}
-	err := u.client.facade.FacadeCall(ctx, "WatchUnitRelations", args, &results)
-	if err != nil {
-		return nil, errors.Trace(apiservererrors.RestoreError(err))
-	}
-	if len(results.Results) != 1 {
-		return nil, errors.Errorf("expected 1 result, got %d", len(results.Results))
-	}
-	result := results.Results[0]
-	if result.Error != nil {
-		return nil, result.Error
+	// To keep the uniter running while the relations domain epic
+	// is underway, temporarily disable relations in the uniter.
+	// JUJU-4856
+
+	return newDisabledRelationWatcher(), nil
+	//var results params.StringsWatchResults
+	//args := params.Entities{
+	//	Entities: []params.Entity{{Tag: u.tag.String()}},
+	//}
+	//err := u.client.facade.FacadeCall(ctx, "WatchUnitRelations", args, &results)
+	//if err != nil {
+	//	return nil, errors.Trace(apiservererrors.RestoreError(err))
+	//}
+	//if len(results.Results) != 1 {
+	//	return nil, errors.Errorf("expected 1 result, got %d", len(results.Results))
+	//}
+	//result := results.Results[0]
+	//if result.Error != nil {
+	//	return nil, result.Error
+	//}
+	//w := apiwatcher.NewStringsWatcher(u.client.facade.RawAPICaller(), result)
+	//return w, nil
+}
+
+func newDisabledRelationWatcher() watcher.StringsWatcher {
+	out := make(chan []string)
+	w := &disabledRelationWatcher{out: out}
+	emptySlice := []string{}
+	w.tomb.Go(func() error {
+		return w.loop(emptySlice)
+	})
+	return w
+}
+
+type disabledRelationWatcher struct {
+	tomb tomb.Tomb
+	out  chan []string
+}
+
+func (d *disabledRelationWatcher) Changes() watcher.StringsChannel {
+	return d.out
+}
+
+func (d *disabledRelationWatcher) loop(changes []string) error {
+	for {
+		select {
+		// Send the initial event or subsequent change.
+		case d.out <- changes:
+		case <-d.tomb.Dying():
+			return nil
+		}
 	}
-	w := apiwatcher.NewStringsWatcher(u.client.facade.RawAPICaller(), result)
-	return w, nil
 }
 
+func (d *disabledRelationWatcher) Kill() {
+	close(d.out)
+}
```

Comment:
You still need to call `d.tomb.Kill(nil)`, otherwise you leak the tomb goroutine.

Also the close should happen as a defer in the loop.